src/architecture/system_model/sim_model.cpp

```diff
@@ -337,8 +332,7 @@ void SimModel::StepUntilStop(uint64_t SimStopTime, int64_t stopPri)
 */
 void SimModel::addNewProcess(SysProcess *newProc)
 {
-    std::vector<SysProcess *>::iterator it;
-    for(it = this->processList.begin(); it != this->processList.end(); it++)
+    for(auto it = this->processList.begin(); it != this->processList.end(); it++)
```

Comment:
We do need the iterator (or something to allow us to refer to a position in the list). Iterators are invalidated by some operations that modify a std::vector (such as insert). However in these two locations we are inserting one element and then exiting the iteration and so never have to care that our insert just invalidated our iterator. It would be nice to find a construct that does need someone to know this at all, but I think his is okay.